app/scripts/directives/service-definition-edit.js

```diff
@@ -46,6 +46,9 @@
             // if sequence type keys have string value, convert them to array
             svalue = convertSeqKeyValueToArray(svalue, skey);
 
+            // if environment key values are a sequence, convert them to a hash array
+            svalue = convertEnvironmentSeqOrHashToArray(svalue, skey);
```

Comment:
In this method and in `convertSeqKeyValueToArray` the `skey` parameter is used to conditionally execute the code.  If you move the conditional check here to surround the function calls on 47 and 50, it will simplify both the methods and make the code easier to understand.
Something like...
```
if (skey === 'environment') {
  svalue = convertEnvironmentSeqOrHashToArray(svalue);
} else if (isKeyTypeSequence(skey)) {
  svalue = convertSeqKeyValueToArray(svalue);
}
```